kernel/src/net/socket/ip/stream/connected.rs

```diff
@@ -45,12 +55,23 @@ impl ConnectedStream {
             bound_socket,
             remote_endpoint,
             is_new_connection,
+            is_receiving_closed: AtomicBool::new(false),
+            is_sending_closed: AtomicBool::new(false),
         }
     }
 
-    pub fn shutdown(&self, _cmd: SockShutdownCmd) -> Result<()> {
-        // TODO: deal with cmd
-        self.bound_socket.close();
+    pub fn shutdown(&self, cmd: SockShutdownCmd, pollee: &Pollee) -> Result<()> {
+        if cmd.shut_read() {
+            self.is_receiving_closed.store(true, Ordering::Relaxed);
+        }
+
+        if cmd.shut_write() {
+            self.is_sending_closed.store(true, Ordering::Relaxed);
+            self.bound_socket.close();
+        }
+
+        self.update_io_events(pollee);
```

Comment:
This can be moved under `if cmd.shut_read()`. For `cmd.shut_write()`, the I/O events will be updated in `poll_ifaces()`.